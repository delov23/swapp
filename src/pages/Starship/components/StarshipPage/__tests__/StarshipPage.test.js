import React from 'react';
import { shallow, mount } from 'enzyme';

import { StarshipPage } from '../';

const EXAMPLE_PROPS = {
  starship: {
    id: 'starships.2',
    name: 'CR90 corvette',
    model: 'CR90 corvette',
    image:
      'https://vignette.wikia.nocookie.net/starwarsmush/images/d/dd/Ship_capital_cr90.jpg/revision/latest?cb=20090605054905',
    starshipClass: 'corvette',
    maxMLPerHour: 60,
    maxAtmosphericSpeed: 950,
    hyperdriveRating: 2,
    crew: 165,
  },
  stData: [
    {
      node: {
        id: 'starships.2',
        cost: 3500000,
        maxAtmosphericSpeed: 950,
        maxMLPerHour: 60,
        hyperdriveRating: 2,
        crew: 165,
      },
    },
  ],
};

describe('<StarshipPage />', () => {
  it('should match snapshot', () => {
    expect(shallow(<StarshipPage {...EXAMPLE_PROPS} />)).toMatchSnapshot();
  });

  it('should display data', () => {
    const wrapper = mount(<StarshipPage {...EXAMPLE_PROPS} />);

    expect(
      wrapper.findWhere(p => p.prop('right') === '0 credits'),
    ).toHaveLength(1);
  });
});
