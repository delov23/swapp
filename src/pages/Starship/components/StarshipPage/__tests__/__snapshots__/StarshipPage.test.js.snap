// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StarshipPage /> should match snapshot 1`] = `
<main>
  <h1
    className="headingLight"
  >
    CR90 corvette
  </h1>
  <h2
    className="subheadingLight"
  >
    (
    CR90 corvette
    )
  </h2>
  <hr />
  <section
    className="content"
  >
    <div
      className="leftBlock"
    >
      <Card
        image="https://vignette.wikia.nocookie.net/starwarsmush/images/d/dd/Ship_capital_cr90.jpg/revision/latest?cb=20090605054905"
        title="CR90 corvette"
      >
        <SpecText
          left="Class"
          right="corvette"
        />
        <SpecText
          left="Cost"
          right="0 credits"
        />
        <SpecText
          left="Crew"
          right={165}
        />
        <SpecText
          left="Max Atmospheric Speed"
          right={950}
        />
        <SpecText
          left="Max ML/h"
          right={60}
        />
        <SpecText
          left="Hyperdrive Rating"
          right={2}
        />
      </Card>
    </div>
    <div
      className="rightBlock"
    >
      <h2
        className="subheadingLight"
      >
        Compared to Starship Class Max
      </h2>
      <div
        className="chartWrapper"
      >
        <div
          className="chart"
        >
          <f
            captions={
              Object {
                "cost": "Cost",
                "crew": "Crew",
                "hyperdriveRating": "HyperD Rat.",
                "maxAtmosphericSpeed": "Max Atm. Speed",
                "maxMLPerHour": "Max ML/h",
              }
            }
            data={
              Array [
                Object {
                  "data": Object {
                    "cost": 0,
                    "crew": 1,
                    "hyperdriveRating": 1,
                    "maxAtmosphericSpeed": 1,
                    "maxMLPerHour": 1,
                  },
                  "meta": Object {
                    "color": "#ffe300",
                  },
                },
              ]
            }
            options={
              Object {
                "axes": false,
                "captionMargin": 10,
                "captionProps": [Function],
                "captions": true,
                "dotProps": [Function],
                "dots": true,
                "scaleProps": [Function],
                "scales": 5,
                "setViewBox": [Function],
              }
            }
            size={400}
          />
        </div>
      </div>
    </div>
  </section>
</main>
`;
